src/message/messageActionSheet.js

```diff
@@ -119,28 +119,33 @@ const deleteMessage = async ({ auth, message, dispatch }) => {
 deleteMessage.title = 'Delete message';
 deleteMessage.errorMessage = 'Failed to delete message';
 
-const markTopicAsRead = async ({ auth, stream, topic, subscriptions }) => {
-  const sub = subscriptions.find(x => x.name === stream);
-  if (sub) {
-    await api.markTopicAsRead(auth, sub.stream_id, topic);
+const markTopicAsRead = async ({ auth, streamId, topic, subscriptions }) => {
+  if (streamId !== undefined) {
+    await api.markTopicAsRead(auth, streamId, topic);
   }
 };
 markTopicAsRead.title = 'Mark topic as read';
 markTopicAsRead.errorMessage = 'Failed to mark topic as read';
 
-const unmuteTopic = async ({ auth, stream, topic }) => {
-  await api.unmuteTopic(auth, stream, topic);
+const unmuteTopic = async ({ auth, streamId, topic, subscriptions }) => {
+  const sub = subscriptions.find(x => x.stream_id === streamId);
```

Comment:
Instead of adding this code, just change `api.unmuteTopic` and `api.muteTopic` to take a stream ID. I'd do that in the commit before this one.

Reply:
done.